slsReceiverSoftware/src/ThreadArping.cpp

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: LGPL-3.0-or-other
+// Copyright (C) 2021 Contributors to the SLS Detector Package
+
+#include "ThreadArping.h"
+#include "sls/container_utils.h"
+#include <iostream>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+ThreadArping::ThreadArping() {}
+
+ThreadArping::~ThreadArping() { StopRunning(); }
+
+pid_t ThreadArping::GetThreadId() const { return threadId; }
+
+bool ThreadArping::IsRunning() const { return runningFlag; }
+
+void ThreadArping::StartRunning() {
+    if (!runningFlag) {
+        if (arpInterfaceIp.size() == 0) {
+            throw sls::RuntimeError("No Interface added to Arping");
+        }
+        runningFlag = true;
+
+        // create thread
+        try {
```

Comment:
Since we are not using pthread directly anywhere else (except calibration) I would try to avoid using it here. You can instead stop the thread by checking for a atomic flag. Something like:

```
#include <thread>
#include <atomic>
#include <chrono>
#include <fmt/core.h>


class ThingWithThread{
    std::atomic<bool> stop_flag;
    std::thread t;

public:
    void StartThread(){
        stop_flag = false;
        t = std::thread(&ThingWithThread::DoStuff, this);
        fmt::print("Started\n");
    }

    void DoStuff(){
        while(!stop_flag){
            fmt::print("Hello\n");
            std::this_thread::sleep_for(std::chrono::seconds(1));
        }
    }

    void StopThread(){
        stop_flag = true;
        t.join();
        fmt::print("Stopped\n");
    }

};

int main(){

    ThingWithThread tt;
    tt.StartThread();
    std::this_thread::sleep_for(std::chrono::seconds(5));
    tt.StopThread();

}
```

Reply:
thanks! will do

Reply:
I cant really do this. I cant wait for the sleep to be done before checking the flag to join with the main  branch.

Reply:
As discussed, will wait for sleep before joining main thread.